woocommerce/woocommerce-android
# [Add coupon] Core feature

Closes: #9470


### Description

This PR adds a core of the coupon creation feature. Before we could view coupon list and edit coupons, but adding new coupon was not possible.

Figma designs: O8MT8FHx8N2zP9pc2namgl-fi-1441%3A13027

##### Not included in this PR:
* Coupon type picker: https://github.com/woocommerce/woocommerce-android/pull/9482 - moved to a separate PR to simplify this PR
* Edit Coupon (coupon creation) screen's title - it's done in the above PR
* Analytics events tracking: #9495

### Testing instructions

1. Go to Coupon list
2. Click on the "+" button
3. Fill new coupon data
4. Tap "Create" and verify the coupon is visible on the coupon list
5. Verify it's present in wp-admin as well and has correct attributes
